### DISCUSSION TITLE
Left-over .shtml files in web folder

### DISCUSSION THREAD

**Discussion by michaelclaesson** [pimcore/pimcore#8182] [Q&A / Support] [answered]
Hi!

We're having an issue with a client's Pimcore installation where every now and then we end up with a number of .shtml files in the web root. Given the name of the temporary files `(Stone_C.tmp-60202b40bfb741.09477288_map.shtml)` we've concluded that it has to do with image thumbnails. The content of the file(s) belongs to the .html file created within the folder of an image thumbnail.

**/var/www/Pimcore/pimroot/web/Stone_C.tmp-60202b40bfb741.09477288_map.shtml**
```
<map id="Stone_C.tmp-60202b40bfb741.09477288" name="Stone_C.tmp-60202b40bfb741.09477288">
  <area href="Stone_C.tmp-60202b40bfb741.09477288_map.shtml" shape="rect" coords="0,0,235,224" alt="" />
</map>
```

**/var/www/Pimcore/pimroot/web/var/tmp/image-thumbnails/Highlights_Puffs/allardsupport/image-thumb__14137__highlight_quarter/Stone_C.html**
```
<?xml version="1.0" encoding="US-ASCII"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html>
<head>
<title>Stone_C.tmp-60202b40bfb741.09477288</title>
</head>
<body style="text-align: center;">
<h1>/var/www/Pimcore/pimroot/web/var/tmp/image-thumbnails/Highlights_Puffs/allardsupport/image-thumb__14137__highlight_quarter/Stone_C.tmp-60202b40bfb741.09477288.html</h1>
<div>
<img usemap="#Stone_C.tmp-60202b40bfb741.09477288" src="/var/www/Pimcore/pimroot/web/var/tmp/image-thumbnails/Highlights_Puffs/allardsupport/image-thumb__14137__highlight_quarter/Stone_C.tmp-60202b40bfb741.09477288.png">
<map id="Stone_C.tmp-60202b40bfb741.09477288" name="Stone_C.tmp-60202b40bfb741.09477288">
  <area href="/var/www/Pimcore/pimroot/web/var/tmp/image-thumbnails/Highlights_Puffs/allardsupport/image-thumb__14137__highlight_quarter/Stone_C.tmp-60202b40bfb741.09477288.html" shape="rect" coords="0,0,235,224" alt="">
</map>
</div>
</body>
</html>
```

Anyone have any idea why these files remain in this folder and don't get deleted? My only guess is that the process that creates the thumbnails and intermediary files crashes mid-execution for some reason and thus the file never gets deleted.

Wondering whether it would be safe to just create a cronjob that deletes these files?

Thanks!

Cheers,
Michael

**Comment by markus-moser** [comment]
Did anyone find a solution for that? It happens in some of our projects too and I do not have any idea why...

**Comment by brusch** [comment, marked as answer]
Seems this comes from ImageMagick, but no idea how to prevent that ... 
https://legacy.imagemagick.org/Usage/montage/#html

**Reply by passioneight** [reply]
Happens for us with **ImageMagick 6.9.10-23**.

Since the domain itself says _legacy_, I guess updating Imagick to `latest` could solve this. Can anyone verify?

**Reply by podarcis** [reply]
> Seems this comes from ImageMagick, but no idea how to prevent that ... https://legacy.imagemagick.org/Usage/montage/#html

We're experiencing this as well - and there are a lot of files. First ever seen on 2022-09-05, maybe a change in Imagemagick version? Currently on `ImageMagick 6.9.7-4 Q16 x86_64 20170114`

My thoughts on this:

- Let's find out which thumbnail transformation or input files leads to those `_map.shtml` relicts. Maybe there's an option to prevent the generation in the first place? Would need to analyze the [source code](https://github.com/ImageMagick/ImageMagick/blob/7.1.0-49/coders/html.c#L411) in case it's not documented.
- Open issue in ImageMagick project?
- Workaround: Remove a `*_map.shtml` if the filename is predictable after successful thumbnail generation.
- Workaround (maybe best?): CHDIR into Pimcore temp directory while performing those operations, so "garbage" pile up there instead of `/web`.
- As @passioneight mentioned: Maybe a more recent ImageMagick version does not behave this way. (Unfortunately not able to test this yet).
